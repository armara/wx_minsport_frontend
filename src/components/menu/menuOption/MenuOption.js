import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

const MenuOption = ({ label, path, pathname, exact }) => {
  return (
    <li>
      <span>
        {((pathname.includes(path) && !exact) || path === pathname) && (
          <i className="black-avatar" />
        )}
      </span>
      <Link to={path}>{label}</Link>
    </li>
  )
}

MenuOption.propTypes = {
  label: PropTypes.string.isRequired,
  pathname: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  exact: PropTypes.bool.isRequired,
}

export default MenuOption
