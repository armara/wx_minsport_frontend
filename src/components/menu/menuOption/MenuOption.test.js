import React from 'react'
import MenuOption from 'components/menu/menuOption/MenuOption'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import { MemoryRouter as Router } from 'react-router-dom'

configure({ adapter: new Adapter() })

describe('<MenuOption />', () => {
  let renderedOption
  const label = 'Test line label'
  const path = 'path'

  beforeEach(() => {
    renderedOption = shallow(
      <MenuOption exact label={label} pathname={path} path={path} />
    )
  })

  it('has correct length', () => {
    expect(renderedOption.length).toEqual(1)
  })

  it('has correct text', () => {
    expect(renderedOption.find('span').text()).toEqual('')
  })

  it('renders correctly', () => {
    const renderedValue = renderer
      .create(
        <Router initialEntries={['/foundations']} initialIndex={0}>
          <MenuOption exact label={label} path={path} pathname={path} />
        </Router>
      )
      .toJSON()
    expect(renderedValue).toMatchSnapshot()
  })
})
