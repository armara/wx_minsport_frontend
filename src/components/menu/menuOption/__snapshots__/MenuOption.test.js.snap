// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuOption /> renders correctly 1`] = `
<li>
  <span>
    <i
      className="black-avatar"
    />
  </span>
  <a
    href="/path"
    onClick={[Function]}
  >
    Test line label
  </a>
</li>
`;
